lib/support/modules/PesterUtil/PesterUtil.psm1

```diff
@@ -0,0 +1,143 @@
+[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12
+
+function Install-ModuleFromNuget {
+    param (
+        $Module,
+        $GalleryUrl         = 'https://www.powershellgallery.com/api/v2'
+    )
+
+    $tempPath       = [io.path]::GetTempPath()
+    $zipFileName    = "{0}.{1}.{2}" -f $Module.Name, $Module.Version, 'zip'
+    $downloadedZip  = Join-Path -Path $tempPath $zipFileName
+    $ModulePath     = Join-Path -Path $PSHome -ChildPath 'Modules'
+    $ModuleFolder   = Join-Path -Path $ModulePath -ChildPath $Module.Name
+
+    if ((Test-Path $ModuleFolder) -and ($PSVersionTable.PSVersion.Major -lt 5 -or $module.Force)) {
+        # Check if available version is correct
+        $ModuleManifest = (Join-Path -Path $ModuleFolder -ChildPath "$($Module.Name).psd1")
+        if ((Test-Path -Path $ModuleManifest) -and -not $Module.Force) {
+            # Import-PowerShellDataFile only exists since 5.1
+            $ManifestInfo = Import-LocalizedData -BaseDirectory (Split-Path -Parent -Path $ModuleManifest) -FileName $Module.Name
+            $ModuleVersionNoPreRelease = $Module.Version -replace '-.*',''
+            # Compare the version in manifest with version required without Pre-release
+            if ($ManifestInfo.ModuleVersion -eq $ModuleVersionNoPreRelease) {
+                Write-Host "Module $($Module.Name) already installed, skipping."
+                return
+            }
+            else {
+                Write-Host "Module $($Module.Name) found with version '$($ManifestInfo.ModuleVersion)', expecting '$ModuleVersionNoPreRelease'."
+            }
+        }
```

Comment:
```suggestion

        if ((Test-Path -Path $ModuleManifest) -and -not $Module.Force) {
            # Import-PowerShellDataFile only exists since 5.1
            $ManifestInfo = Import-LocalizedData -BaseDirectory (Split-Path -Parent -Path $ModuleManifest) -FileName $Module.Name
            $ModuleVersionNoPreRelease = $Module.Version -replace '-.*'

            # Compare the version in manifest with version required without Pre-release
            if ($ManifestInfo.ModuleVersion -eq $ModuleVersionNoPreRelease) {
                Write-Host "Module $($Module.Name) already installed, skipping."
                return
            }
            else {
                Write-Host "Module $($Module.Name) found with version '$($ManifestInfo.ModuleVersion)', expecting '$ModuleVersionNoPreRelease'."
            }
        }
```